kythe/rust/extractor/src/bin/save_analysis.rs

```diff
@@ -38,10 +38,12 @@ fn generate_arguments(arguments: Vec<String>, output_dir: &Path) -> Result<Vec<S
         .position(|arg| arg == "--")
         .ok_or_else(|| anyhow!("Could not find the start of the rustc arguments"))?;
 
-    // Keep the "--" argument and replace it with an empty string because
+    // The rust compiler executable path is the argument directly after "--"
+    // We keep the path argument and  replace it with an empty string because
     // `kythe_rust_extractor::generate_analysis` requires the first argument in
     // `arguments` to be an empty string
-    let mut rustc_arguments = arguments.split_at(argument_position).1.to_vec();
+    let mut rustc_arguments = arguments.split_at(argument_position + 1).1.to_vec();
```

Comment:
So it's not clear to me where this logic belongs. kythe_rust_extractor::generate_analysis could be changed to do this, moving the oddity closer to where it's needed. Why do it at the extractor?

Reply:
The main reason this was broken out was to keep the library build-system agnostic agnostic, with binary clients performing the pre-processing of the Rust compiler arguments. Initially this was done to make it easier for Fuchsia's extractor, but they seemingly don't pull in the extractor library anyways. I can definitely move this logic into the library if you believe that's the best course of action.

Reply:
It's not clear that either place is better, since they're both in the extractor.